src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -108,6 +109,16 @@ public RestClientMethod BuildRequestMethod(Operation operation, ServiceRequest s
             );
         }
 
+        public static PathSegment[] BuildRequestPathSegments(Operation operation, ServiceRequest serviceRequest, HttpRequest httpRequest, RestClientBuilder restClientBuilder)
+        {
+            var allParameters = restClientBuilder.GetOperationAllParameters(operation, operation.Parameters);
+            var methodParameters = restClientBuilder.BuildMethodParameters(allParameters);
+            var references = allParameters.ToDictionary(kvp => GetRequestParameterName(kvp.Key), kvp => new ParameterInfo(kvp.Key, restClientBuilder.CreateReference(kvp.Key, kvp.Value)));
+            var buildContext = new RequestMethodBuildContext(methodParameters, references);
+            BuildRequestParameters(httpRequest, buildContext, out var queryParams, out var headerParams, out var uriParams, out var pathParams);
```

Comment:
This method creates lots of objects that are just thrown away. However, from what I understand, `PathSegment` instances are not needed. `PathSegment.ParsePathSegment` uses only `PathSegment.Value` and `PathSegment.Escape`. The first one is provided by `CreateReference`, the last one is simply `!requestParameter.Extensions!.SkipEncoding`. So something like:

```cs
 public IReadOnlyDictionary<string, ParameterInfo> GetReferencesToOperationParameters(Operation operation, IEnumerable<RequestParameter> requestParameters)
{
    var allParameters = GetOperationAllParameters(operation, requestParameters);
    return allParameters.ToDictionary(kvp => GetRequestParameterName(kvp.Key), kvp => new ParameterInfo(kvp.Key, CreateReference(kvp.Key, kvp.Value)));
}
```

should be enough.